sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/internal/ComponentProvider.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure.spi.internal;
+
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+
+/**
+ * Provides configured instances of SDK extension components. {@link ComponentProvider} allows SDK
+ * extension components which are not part of the core SDK to be referenced in file based
+ * configuration.
+ *
+ * @param <T> the type of the SDK extension component. See {@link #getType()}.
+ */
+// TODO (jack-berg): list the specific types which are supported in file configuration
```

Comment:
This PR or a follow-up?

Reply:
Followup. As of now, this SPI is only used to load `PrometheusHttpServer`. Need to go through and add support for custom SDK extension components (SpanExporter, MetricExporter, Sampler, etc) one by one.